### ISSUE TITLE
Evocrash refactoring

### ISSUE THREAD

**Issue by danzone** [STAMP-project/EvoCrash#14]
#### Characteristics
- **Issue Type**: feature
- **Severity**: major
- **Tool/Service/Component**: Evocrash 1.0.3

#### Description
In order to develop plugins and industrialize Evocrash, a refactoring is needed. Evocrash depends on several Evosuite components that actually are part of Evocrash code base itself. A first step towards Evocrash refactoring is to identify, within Evocrash code base, which Evosuite packages aren't (or won't be) affected by Evocrash changes, remove them, and add them as Maven dependencies within POM file. After that, Evocrash specific packages can be renamed from `org.evosuite.*` to `org.evocrash.*`.
In short:
- [ ] identify Evosuite packages (i.e. Evocrash packages which are (and will remain) identical to corresponding Evosuite ones
- [ ] replace these packages, one at a time, with the corresponding Evosuite Maven dependency
- [ ] rename Evocrash specific packages as `org.evocrash.*`

ENG can help in this activity.

**Comment by danzone** [comment]
According to the first step, I've just inspected Evosuite packages available within Maven Repositories (https://mvnrepository.com/artifact/org.evosuite):
- [ ] Evosuite Master (`org.evosuite.*` and several other packages: `ch.*`, `javax.*`, etc)
- [ ] Evosuite Runtime (`org.evosuite.annotations` and `org.evosuite.runtime`)
- [ ] Evosuite Client (several `org.evosuite.*` packages and others)
- [ ] Evosuite Standalone runtime
- [ ] Evosuite plugin (several `org.evosuite.*` sub-projects, one each for a specific plugin)
- [ ] Evosuite Shaded (several `org.evosuite.*` packages and others)
- [ ] Evosuite Generated
- [ ] Evosuite

Inspecting those packages, it seems that all the Evocrash-related code stands within client package: is it correct?

**Comment by pderakhshanfar** [comment]
@danzone EvoCrash uses the complete package of EvoSuite. client package cannot work without the master client.

**Comment by danzone** [comment]
Hi @pderakhshanfar, yes, I see. I'm wondering if we can simply add Evosuite as a dependency within Evocrash POM file, removing it from Evocrash code base. In this way, Evocrash maintenance and evolution will be easier, and we won't need to take care about Evosuite evolution. Otherwise we would be forced to take with Evocrash also all the Evosuite code, and mantain it (bringing back from Evosuite master code base all the fixes, for instance). What do you think about it?

**Comment by pderakhshanfar** [comment]
Hi, @danzone. Yes, we have the same plan to use EvoSuite as a library in the refactored version.

**Comment by danzone** [comment]
Excellent! Have you any roadmap about the refactoring? I think integration activities would benefit from it

**Comment by pderakhshanfar** [comment]
The roadmap is not 100% clear yet. We should have it ASAP.

**Comment by danzone** [comment]
OK, I'm looking forward for it. Thank you @pderakhshanfar !

**Comment by danzone** [comment]
HI guys! @luandrea  needs to start working hardly on Evocrash Maven plugin. Is the refactoring complete? If not, is the state-of-the-art stable enough to start the development? After the mid-project review the integration activities became more critical than in the past, and th Deliverable 4.3 is upcoming (end of November, but of course we have to do a lot of other activities).

Please let us know, we need to define the roadmap.

**Comment by xdevroey** [comment]
Hi @danzone @luandrea! Refactoring is ongoing, but a first alpha version is available at https://github.com/STAMP-project/botsing/ (since we reimplemented everything, we also changed the name of the tool). I'll open a 'feature' issue there about the Maven plugin.  See https://github.com/STAMP-project/botsing/issues/4
